docs: improve configuration templates (#7863)
renovatebot/renovate: docs/usage/configuration-templates.md

`branchTopic` depends on the package manager and upgrade type, so you will see a lot of variety.
This is probably a setting you want to change yourself.
Be careful, and consider posting your config to https://github.com/renovateapp/config-help first to get help from the Renovate team with your config.

## Commit Message

Renovate will use one commit per branch, this makes it easy for you to merge.
As such, the `commitMessage` reflects the contents of the branch and is usually the same as the PR title.

`commitMessage` has a default value of `{{commitMessagePrefix}} {{commitMessageAction}} {{commitMessageTopic}} {{commitMessageExtra}} {{commitMessageSuffix}}`, with the intention that you only edit some of those subcomponents.

You usually don't need to edit `commitMessagePrefix`, this option is used by Renovate if it needs to add a prefix to conform to the Semantic Commit convention.
Do not touch this unless you know what you're doing.

`commitMessageAction` is usually just one word, e.g. 'Update', 'Pin', 'Refresh', etc.
You're probably fine leaving this setting alone, though you can change it.

> e.g. if you prefer that Renovate uses the term 'Upgrade' instead of 'Update'
> TODO: Explain how user can change `commitMessageAction` for 'Update' -> 'Upgrade'

Reviewer: ```suggestion
e.g. if you prefer that Renovate uses the term 'Upgrade' instead of 'Update' then you could configure `"commitMessageAction": "Upgrade"`.
```

`commitMessageTopic` is usually two to three words that identify _what_ is being updated.
e.g. it might be `dependency react` or `Docker image ubuntu`.
You may want to edit this.

> If you think your new `commitMessageTopic` is helpful for others, please open a PR at (TODO: INSERT LOCATION WHICH ACCEPTS PR).

Reviewer: ```suggestion
If you think your new `commitMessageTopic` is helpful for others, please [open a PR](https://github.com/renovatebot/renovate/pulls).
```

> You can also add this to the preset config repo (TODO: INSERT LINK TO PRESET CONFIG REPO) so that others with similar requirements can benefit too.

Reviewer: ```suggestion
```

`commitMessageExtra` refers to the version being updated to.
e.g. `to v16` for a major upgrade, or `to v16.0.3` for a patch update.
It can be empty in some cases, like if the action/topic doesn't change a package version, e.g. `Pin Docker digests`.

> `commitMessageSuffix` defaults to empty and is there for flexibility and future use (TODO: EXPLAIN IF THIS IS FOR USER'S BENEFIT OF FOR RENOVATE'S BENEFIT).

Reviewer: ```suggestion
`commitMessageSuffix` defaults to empty but is currently used in two cases:
- Differentiating major from non-major groups
- Differentiating between PRs from different base branches
```

Maybe for `major` updates you always want the PR to end with `(MAJOR)`, for instance.

`commitBody` is used if you wish to add multi-line commit messages, such as for the `Signed-off-by` fields, or adding `[skip-ci]`, etc.
It is appended to the generated `commitMessage`, separated by a newline.

## PR Title

Because commit messages match with the PR title, the PR title template defaults to `null` and inherits/copies the value from `commitMessage`.

> If you have a requirement where `prTitle` should be different from `commitMessage`, then please raise a feature request (TODO: INSERT LINK TO RENOVATE REPO) for discussion.

Reviewer: ```suggestion
If you have a requirement where `prTitle` should be different from `commitMessage`, then please [raise a feature request](https://github.com/renovatebot/renovate/issues) for discussion.
```

## PR Body

> The PR Body is currently a little difficult to edit because of its size, however it should soon be redesigned like the other templates to allow more easier editing without needing to copy/paste the whole template.
> TODO: CHECK WITH RENOVATE TEAM IF THE PR BODY IS USER EDITABLE NOW

Reviewer: ```suggestion
The PR body can be edited multiple ways:
- Editing the entire layout/flow using `prBodyTemplate` (but try to avoid that if possible)
- Adding header or footer using `prHeader` and `prFooter`
- Adding notes using `prBodyNotes`
- Editing the embedded table using `prBodyDefinitions` and `prBodyColumns`
```

> TODO: CHECK WITH RENOVATE TEAM IF THE PR BODY IS USER EDITABLE NOW

Author: > TODO: CHECK WITH RENOVATE TEAM IF THE PR BODY IS USER EDITABLE NOW

@rarkins can a user edit the PR body text?

It can be empty in some cases, like if the action/topic doesn't change a package version, e.g. `Pin Docker digests`.

`commitMessageSuffix` defaults to empty but is currently used in two cases:

- Differentiating major from non-major groups
- Differentiating between PRs from different base branches, maybe for `major` updates you always want the PR to end with `(MAJOR)`, for instance

`commitBody` is used if you wish to add multi-line commit messages, such as for the `Signed-off-by` fields, or adding `[skip-ci]`, etc.
It is appended to the generated `commitMessage`, separated by a newline.

## PR Title

Because commit messages match with the PR title, the PR title template defaults to `null` and inherits/copies the value from `commitMessage`.
If you have a requirement where `prTitle` should be different from `commitMessage`, then please [raise a feature request](https://github.com/renovatebot/renovate/issues) for discussion.

## PR Body

You can change the PR body in the following ways:

- Change the entire layout/flow by using `prBodyTemplate` (we do not recommend this)
- Add a header by using `prHeader`
- Add a footer by using `prFooter`
- Add a note by using `prBodyNotes`

> - Edit the embedded table by using `prBodyDefinitions` and `prBodyColumns`

Author: @rarkins I've used your text and made some changes, let me know if you like those or not. 😉
